sstables/generation_type.cc

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "sstables/generation_type.hh"
+#include "api/api.hh"
+#include <cctype>
+#include <fmt/core.h>
+#include <cstdint>
+#include <iterator>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace {
+sstring to_base36(uint64_t i) {
+    std::vector<char> buf(13, '\0');
+    auto s = buf.begin();
+    do {
+        *s = "0123456789abcdefghijklmnopqrstuvwxyz"[i % 36];
+        i /= 36;
+        s++;
+    } while (i);
+
+    return sstring(std::make_reverse_iterator(s), buf.rend());
+}
+
+uint64_t from_base36(const sstring& buf) {
+    uint64_t total = 0;
+    for (auto i = buf.size() - 1; i >= 0; --i) {
+        total *= 36;
+        const char c = std::tolower(buf[i]);
+        total += (c >= 'a' && c <= 'z') ? c - 'a' + 10 : c - '0';
+    }
+    return total;
+}
+} // anon. namespace
```

Comment:
there's a base64 impl in utils/base64.hh. recommend doing the same for base36 and using a similar interface.